lib/internal/streams/from.js

```diff
@@ -53,18 +50,22 @@ function from(Readable, iterable, opts) {
   };
 
   readable._destroy = function(error, cb) {
-    if (needToClose) {
-      needToClose = false;
-      close().then(
-        () => process.nextTick(cb, error),
-        (e) => process.nextTick(cb, error || e),
-      );
-    } else {
-      cb(error);
-    }
+    close(error).then(
+      () => process.nextTick(cb, error), // nextTick is here in case cb throws
+      (e) => process.nextTick(cb, error || e),
+    );
   };
 
-  async function close() {
+  async function close(error) {
+    const hadError = typeof error !== 'undefined';
```

Comment:
Wouldn't `error` be `null`?

https://github.com/nodejs/node/blob/683754c6f3ea8e04d34b2dad854aca95601c14a8/lib/internal/streams/destroy.js#L67